models/Posts.js

```diff
@@ -458,4 +479,29 @@ PostSchema.methods.addMedia = function (type, data) {
   });
 };
 
+PostSchema.methods.getMedia = function (type) {
+  return new Promise((resolve, reject) => {
+    Media.findOne({ _id: this.media._id })
+      .then((media) => {
+        resolve(media.getMedia(type));
+      }).catch((err) => {
+        reject(err);
+      });
+  });
+};
+
+PostSchema.methods.removeMedia = function () {
```

Comment:
So this doesn't actually delete the Media object, but just removes it from the post. Is this what's intended?

Reply:
ehhh... not really, but it does work. I'll look into how easy it is to delete the object, since space does matter for us

Reply:
If I were to guess it would just be one extra line. something along the lines of `getMedia.delete().then( ... )`